CMakeLists.txt

```diff
@@ -0,0 +1,160 @@
+		option(DATE_TZ_NO_REMOTE "Disable using cURL to download IANA Time Zone rules" On)
+	endif()
+	if(WIN32)
+		option(DATE_TZ_NO_TZMAP  "Disable Time Zone name mapping on Windows" Off)
+	else()
+		option(DATE_TZ_NO_TZMAP  "Disable Time Zone name mapping on Windows" On)
+	endif()
+	if(DATE_TZ_SHELL_REMOVE_DIRECTORY AND DATE_TZ_SHELL_REMOVE_FILE AND DATE_TZ_SHELL_MOVE_FILE AND DATE_TZ_SHELL_UNTAR AND DATE_TZ_SHELL_MAKE_DIRECTORY)
+		option(DATE_TZ_NO_DATE_TZ_SHELL  "Disable using shell programs to manage IANA Time Zone rules" Off)
+	else()
+		option(DATE_TZ_NO_DATE_TZ_SHELL  "Disable using shell programs to manage IANA Time Zone rules" On)
+	endif()
+endif()
+
+message(STATUS "Using ${CMAKE_CXX_COMPILER} (compiler id: ${CMAKE_CXX_COMPILER_ID})")
+if ("${CMAKE_CXX_COMPILER_ID}" MATCHES ".*Clang")
+	if(NOT "${CMAKE_CXX_FLAGS}" MATCHES "-stdlib")
+		set(CMAKE_CXX_FLAGS "-std=c++14 -stdlib=libc++ ${CMAKE_CXX_FLAGS}")
+	endif()
+endif ()
+if(NOT NEED_PARENT)
+	if("${CMAKE_CXX_COMPILER_ID}" MATCHES ".*Clang" OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Gnu")
+		set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Werror")
+	endif()
+endif()
+
+if(NEED_PARENT)
+	set(DATE_INCLUDES ${CMAKE_CURRENT_SOURCE_DIR} PARENT_SCOPE)
+endif()
+
+if(NOT DATE_NO_TZ)
+	add_definitions(-DDATE_BUILD_LIB=1)
+
+	if(WIN32)
+		if(DATE_TZ_NO_TZMAP)
+			add_definitions(-DTIMEZONE_MAPPING=0)
+		else()
+			add_definitions(-DTIMEZONE_MAPPING=1)
+		endif()
+	endif()
+	if(DATE_TZ_NO_LAZY)
+		add_definitions(-DLAZY_INIT=0)
+	else()
+		add_definitions(-DLAZY_INIT=1)
+	endif()
+
+	set(DATE_INC ${CMAKE_CURRENT_SOURCE_DIR})
+
+	if(NOT DATE_TZ_NO_REMOTE AND CURL_FOUND)
+		message(STATUS "Date: Found cURL. Explicitly enabling HAS_REMOTE_API")
+		add_definitions(-DHAS_REMOTE_API=1)
+		add_definitions(-DAUTO_DOWNLOAD=1)
+		set(DATE_INC ${DATE_INC} ${CURL_INCLUDE_DIRS})
+		set(TZLINKS ${CURL_LIBRARIES})
+	else()
+		message(WARNING "Date: Disabling auto download")
+		add_definitions(-DHAS_REMOTE_API=0)
+		add_definitions(-DAUTO_DOWNLOAD=0)
+		set(TZLINKS "")
+	endif()
+
+	add_library(date-tz STATIC tz.cpp)
+
+	target_link_libraries(date-tz ${TZLINKS})
+
+	set_property(TARGET date-tz PROPERTY CXX_STADARD 14)
+
+	if(NEED_PARENT)
+		set(DATE_TZ_INCLUDES ${DATE_INC} PARENT_SCOPE)
+		set(DATE_TZ_LIBS ${TZLINKS} date-tz PARENT_SCOPE)
+	else()
+		set(DATE_TZ_LIBS ${TZLINKS} date-tz)
+	endif()
+
+	if(DATE_ENABLE_TESTS)	
+		enable_testing()
+
+		include_directories(${CMAKE_SOURCE_DIR} ${CMAKE_SOURCE_DIR}/test)
+		add_executable(validate test/tz_test/validate.cpp)
+
+		target_link_libraries(validate ${DATE_TZ_LIBS})
+		
+		add_test(validate validate)
+		if(NOT MSVC)
+			set(DATE_TZ_DO_FAIL_PASS On)
+		elseif(${MSVC_VERSION} GREATER 1910)
+			set(DATE_TZ_DO_FAIL_PASS On)
+		endif()
+		if(DATE_TZ_DO_FAIL_PASS)
+			file(GLOB_RECURSE DATE_STATIC_TESTS RELATIVE ${CMAKE_SOURCE_DIR}/test/ *.fail.cpp)
+			file(GLOB_RECURSE DATE_RUNTIME_TESTS RELATIVE ${CMAKE_SOURCE_DIR}/test/ *.pass.cpp)
```

Comment:
GLOB_RECURSE is considered bad practice because it requires invoking cmake again to update the filelist, which can be problematic with version control.